src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -281,15 +297,124 @@ namespace AppInstaller::CLI::Workflow
                 isDirectoryCreated = true;
             }
 
-            Portable::PortableARPEntry uninstallEntry = Portable::PortableARPEntry(
-                ConvertToScopeEnum(context.Args.GetArg(Execution::Args::Type::InstallScope)),
-                context.Get<Execution::Data::Installer>()->Arch,
-                ConvertToUTF16(GetPortableProductCode(context)));
-            uninstallEntry.SetValue(PortableValueName::InstallDirectoryCreated, isDirectoryCreated);
+            if (std::filesystem::exists(targetFullPath))
+            {
+                std::filesystem::remove(targetFullPath);
+                AICLI_LOG(CLI, Info, << "Removing existing portable exe at: " << targetFullPath);
+            }
 
             Filesystem::RenameFile(installerPath, targetFullPath);
             AICLI_LOG(CLI, Info, << "Portable exe moved to: " << targetFullPath);
 
+            if (!uninstallEntry[PortableValueName::InstallDirectoryCreated].has_value())
+            {
+                uninstallEntry.SetValue(PortableValueName::InstallDirectoryCreated, isDirectoryCreated);
```

Comment:
Why not just always set it?

Reply:
Since upgrade and install use the same flow, if we always set it, then any upgrade after the first install will always overwrite this value to false because the directory now exists, even if we were the ones who created it during the first install. I think we should try to honor the initial set value because that should be set when the package is first installed.